### ISSUE TITLE
installK3.sh ERROR: argument --resource-group/-g: expected one argument

### ISSUE THREAD

**Issue by colinrippeyfinarne** [microsoft/azure_arc#1604]
I'm running the DevOps scenario and getting an issue inside the script that runs as part of the VM extension for the k3s.

From what I can tell the installK3.sh script has an issue on line 72

resourceGroup=$(sudo -u $adminUsername az resource list --query "[?name=='$vmName']".[resourceGroup] --resource-type "Microsoft.Compute/virtualMachines" -o tsv)

If I try to run this command in isolation:

az resource list --query "[?name=='ArcBox-K3s-<myguid>']" -o tsv

I see a result ok, but if I try to append the .[resourceGroup]:

az resource list --query "[?name=='ArcBox-K3s-<myguid>']".[resourceGroup] -o tsv

I get the following error:

ParserError: 
Line |
   1 |  … "[?name=='ArcBox-K3s-,myguid.']".[resourc …
     |                                                                 ~
     | Missing property name after reference operator.

I'm guessing this error is swallowed and it means that the resourceGroup variable is empty and the next line in the script fails with the error: 

ERROR: argument --resource-group/-g: expected one argument

A proposed fix would be to change the query to:

az resource list --query "[?name=='ArcBox-K3s-<myguid>'].resourceGroup" -o tsv

Running this in isolation gives me the correct resource group name.

I'll try to run the remaining commands in the script manually to see if I can successfully complete the deployment.

**Comment by likamrat** [comment]
@colinrippeyfinarne can you please share your deployment command and parameters file? Make sure to mask your service principal secret.

**Comment by colinrippeyfinarne** [comment]
Hi @likamrat 

I've successfully redeployed today, 

Best to clear up my "wrong assumptions" above.

First, I was missing granting the service principal the Owner role on the subscription (I was reusing a service principal from previous deployments but this time on a different subscription so had not granted it the owner role).

Second, In my "debugging" I was manually running some of the az commands in the installK3s.sh script (and to confuse matters was looking inside the "wrong" installK3s.sh file).

I can now see that the VM ext on the ArcBox-K3s uses the script from:

azure_arc/azure_jumpstart_arcbox/artifacts/installK3s.sh

But...I was running the individual az commands in a pwsh shell so got the error on the .[resourceGroup] property and thought "that must be the issue, the $resourceGroup variable must be empty".

I then realised that the installK3s.sh "is a bash script" doh!!! and when I switched to a bash shell the commands worked.

Hence my confusion...the resourceGroup is there and is returned successfully "if" the service principal can see it but in my case I had not granted the service principal access to the different subscription!

So now...I have made sure the service principal does the Owner role.

First attempt failed again with seemingly the same error - ERROR argument --resource-group. 

This issue was my service principal now had access to two subscriptions and unfortunately the default subscription is NOT the one I am deploying to so I think the installK3s.sh script fails on retrieving the resourceGroup as the current subscription does not have any of the resources.

Perhaps a tweak to the installK3s.sh file to set the account after az login command to be sure the session is using the correct subscription...if I had followed your instructions to the letter this would not be required of course as I would be using a brand new service principal that will only have access to one subscription.

I then removed the service principal's access to the other subscription, re-ran and now I get everything working as expected.

I always think the best way to learn is to break stuff...and I broke a few different things here but I have at least learned "there's a difference in pwsh and bash syntax" as if I didn't already know lol.
